x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -224,6 +227,55 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
     return await this.baseClient.delete(type, id, options);
   }
 
+  public async bulkDelete(
+    objects: SavedObjectsBulkDeleteObject[],
+    options: SavedObjectsBulkDeleteOptions
+  ): Promise<SavedObjectsBulkDeleteResponse> {
+    // check if the user is authorized to bulk_delete the object types, exit early if not allowed to
+    try {
+      const args = { objects, options };
+      await this.legacyEnsureAuthorized(
+        this.getUniqueObjectTypes(objects),
+        'bulk_delete',
+        options?.namespace,
+        {
+          args,
+        }
+      );
+    } catch (error) {
+      objects.forEach(({ type, id }) =>
+        this.auditLogger.log(
+          savedObjectEvent({
+            action: SavedObjectAction.DELETE,
+            savedObject: { type, id },
+            error,
+          })
+        )
+      );
+      throw error;
+    }
+    // call the core saved objects client
+    const response = await this.baseClient.bulkDelete(objects, options);
+    // log the result of each object's delete outcome
+    response?.statuses.forEach(({ id, type, success, error }) => {
+      if (!error) {
+        const auditEventOutcome = success === true ? 'success' : 'failure';
+        this.auditLogger.log(
+          savedObjectEvent({
+            action: SavedObjectAction.DELETE,
+            savedObject: { type, id },
+            outcome: auditEventOutcome,
+            error: error ? error : undefined,
+          })
+        );
+      }
+    });
+    // the response only contains saved objects' type and id and there's no direct object namespace to delete.
+    // However, the id might contain a reference to a space and we may need to redact that.
+    // FOLLOW UP: figure out how to remove the saved object space from the id
```

Comment:
Hum, aren't we converting the 'technical' ID to the 'functional' one within the `SOR.bulkDelete` implementation?